### PR TITLE
python3Packages.faker: 4.18.0 -> 5.5.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 5.5.1.

Changelog: https://github.com/joke2k/faker/blob/master/CHANGELOG.md

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/faker/default.nix [hunk 1]
```diff
@@ -1,59 +1,45 @@
...
     ukpostcodeparser
     validators
-  ]
-  ++ lib.optionals (pythonOlder "3.3") [ mock ]
-  ++ lib.optionals (pythonOlder "3.0") [ more-itertools ];
+  ];
 
   # avoid tests which import random2, an abandoned library
   pytestFlagsArray = [
     "--ignore=tests/providers/test_ssn.py"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/faker/default.nix:35]: Did you check if this still applies?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109269` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
11 packages marked as broken and skipped:
</summary>
<ul>
<li>gscan2pdf</li>
<li>mailman-web</li>
<li>odpdown</li>
<li>python37Packages.mailman-web</li>
<li>python37Packages.rl-coach</li>
<li>python37Packages.toggl-cli</li>
<li>python38Packages.mailman-web</li>
<li>python38Packages.rl-coach</li>
<li>python38Packages.toggl-cli</li>
<li>python39Packages.mailman-web</li>
<li>python39Packages.toggl-cli</li>
</ul>
</details>
<details>

  <summary>13 packages failed to build and are new build failures:</summary>
  <ul>
<li>prospector: log was empty</li>
<li>python37Packages.django_extensions: log <a href="https://termbin.com/6m9h">https://termbin.com/6m9h</a></li>
<li>python37Packages.django_silk: log <a href="https://termbin.com/hijb">https://termbin.com/hijb</a></li>
<li>python37Packages.pylint-django: log <a href="https://termbin.com/oxd0v">https://termbin.com/oxd0v</a></li>
<li>python37Packages.qiskit: log was empty</li>
<li>python37Packages.qiskit-aqua: log was empty</li>
<li>python38Packages.django_extensions: log <a href="https://termbin.com/kqvj">https://termbin.com/kqvj</a></li>
<li>python38Packages.django_silk: log <a href="https://termbin.com/e0b1">https://termbin.com/e0b1</a></li>
<li>python38Packages.pylint-django: log <a href="https://termbin.com/8jz5z">https://termbin.com/8jz5z</a></li>
<li>python38Packages.qiskit: log was empty</li>
<li>python38Packages.qiskit-aqua: log was empty</li>
<li>python39Packages.django_extensions: log <a href="https://termbin.com/h1hi">https://termbin.com/h1hi</a></li>
<li>python39Packages.django_silk: log <a href="https://termbin.com/6s98">https://termbin.com/6s98</a></li>
  </ul>
</details>
<details>
<summary>
72 packages built:
</summary>
<ul>
<li>flexget</li>
<li>noto-fonts-emoji</li>
<li>photon</li>
<li>python37Packages.Quandl (python37Packages.quandl)</li>
<li>python37Packages.afdko</li>
<li>python37Packages.booleanoperations</li>
<li>python37Packages.defcon</li>
<li>python37Packages.factory_boy</li>
<li>python37Packages.faker</li>
<li>python37Packages.flask-restx</li>
<li>python37Packages.fontmath</li>
<li>python37Packages.fontparts</li>
<li>python37Packages.fontpens</li>
<li>python37Packages.fonttools</li>
<li>python37Packages.minio</li>
<li>python37Packages.mutatormath</li>
<li>python37Packages.nototools</li>
<li>python37Packages.psautohint</li>
<li>python37Packages.pytest-factoryboy</li>
<li>python37Packages.pytest-randomly</li>
<li>python37Packages.scrapy-fake-useragent</li>
<li>python37Packages.sslyze</li>
<li>python37Packages.tld</li>
<li>python37Packages.ufoprocessor</li>
<li>python38Packages.Quandl (python38Packages.quandl)</li>
<li>python38Packages.afdko</li>
<li>python38Packages.booleanoperations</li>
<li>python38Packages.defcon</li>
<li>python38Packages.factory_boy</li>
<li>python38Packages.faker</li>
<li>python38Packages.flask-restx</li>
<li>python38Packages.fontmath</li>
<li>python38Packages.fontparts</li>
<li>python38Packages.fontpens</li>
<li>python38Packages.fonttools</li>
<li>python38Packages.minio</li>
<li>python38Packages.mutatormath</li>
<li>python38Packages.nototools</li>
<li>python38Packages.psautohint</li>
<li>python38Packages.pytest-factoryboy</li>
<li>python38Packages.pytest-randomly</li>
<li>python38Packages.scrapy-fake-useragent</li>
<li>sslyze (python38Packages.sslyze)</li>
<li>python38Packages.tld</li>
<li>python38Packages.ufoprocessor</li>
<li>python39Packages.afdko</li>
<li>python39Packages.booleanoperations</li>
<li>python39Packages.defcon</li>
<li>python39Packages.factory_boy</li>
<li>python39Packages.faker</li>
<li>python39Packages.flask-restx</li>
<li>python39Packages.fontmath</li>
<li>python39Packages.fontparts</li>
<li>python39Packages.fontpens</li>
<li>python39Packages.fonttools</li>
<li>python39Packages.minio</li>
<li>python39Packages.mutatormath</li>
<li>python39Packages.nototools</li>
<li>python39Packages.psautohint</li>
<li>python39Packages.pytest-factoryboy</li>
<li>python39Packages.pytest-randomly</li>
<li>python39Packages.scrapy-fake-useragent</li>
<li>python39Packages.sslyze</li>
<li>python39Packages.tld</li>
<li>python39Packages.ufoprocessor</li>
<li>rictydiminished-with-firacode</li>
<li>scfbuild</li>
<li>sourcehut.gitsrht</li>
<li>sourcehut.todosrht</li>
<li>visidata</li>
<li>xits-math</li>
<li>xquartz</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.ufoprocessor:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.ufoprocessor:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.ufoprocessor:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>

File: pkgs/development/python-modules/faker/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/development/python-modules/faker/default.nix:35]: Just checked. Still an issue.

**Comment by Mic92** [comment]
It is weired that your django builds get `PermissionError: [Errno 1] Operation not permitted: '/usr/share/zoneinfo'`

**Comment by Mic92** [comment]
I suppose this has something to do with darwin's sandbox? cc @LnL7 ?

**Comment by SuperSandro2000** [comment]
> I suppose this has something to do with darwin's sandbox? cc @LnL7 ?

Yes, it is. 

Also I can't build this PR on linux because it tries to build the kernel.

File: pkgs/development/python-modules/faker/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/faker/default.nix:35]: Unfortunately, [yes](https://github.com/joke2k/faker/blob/master/tests/providers/test_ssn.py#L10).

**Comment by Mic92** [comment]
When building only python packages this is the result:

Result of `nixpkgs-review pr 109269` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.rl-coach</li>
    <li>python37Packages.skein</li>
    <li>python37Packages.zake</li>
    <li>python38Packages.rl-coach</li>
    <li>python38Packages.skein</li>
    <li>python38Packages.zake</li>
    <li>python39Packages.skein</li>
    <li>python39Packages.zake</li>
  </ul>
</details>
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>python37Packages.foundationdb60</li>
    <li>python37Packages.foundationdb61</li>
    <li>python37Packages.pylint-django</li>
    <li>python37Packages.slither-analyzer</li>
    <li>python38Packages.foundationdb60</li>
    <li>python38Packages.foundationdb61</li>
    <li>python38Packages.pylint-django</li>
    <li>python38Packages.slither-analyzer</li>
    <li>python39Packages.django_silk</li>
    <li>python39Packages.foundationdb60</li>
    <li>python39Packages.foundationdb61</li>
    <li>python39Packages.slither-analyzer</li>
  </ul>
</details>
<details>
  <summary>95 packages built:</summary>
  <ul>
    <li>python37Packages.Quandl (python37Packages.quandl)</li>
    <li>python37Packages.afdko</li>
    <li>python37Packages.booleanoperations</li>
    <li>python37Packages.cnvkit</li>
    <li>python37Packages.defcon</li>
    <li>python37Packages.django_extensions</li>
    <li>python37Packages.django_silk</li>
    <li>python37Packages.factory_boy</li>
    <li>python37Packages.faker</li>
    <li>python37Packages.flask-restx</li>
    <li>python37Packages.fontmath</li>
    <li>python37Packages.fontparts</li>
    <li>python37Packages.fontpens</li>
    <li>python37Packages.fonttools</li>
    <li>python37Packages.foundationdb51</li>
    <li>python37Packages.foundationdb52</li>
    <li>python37Packages.hyperkitty</li>
    <li>python37Packages.kazoo</li>
    <li>python37Packages.mailman-web</li>
    <li>python37Packages.minio</li>
    <li>python37Packages.monosat</li>
    <li>python37Packages.mutatormath</li>
    <li>python37Packages.nototools</li>
    <li>python37Packages.psautohint</li>
    <li>python37Packages.pytest-factoryboy</li>
    <li>python37Packages.pytest-randomly</li>
    <li>python37Packages.qiskit</li>
    <li>python37Packages.qiskit-aqua</li>
    <li>python37Packages.rpy2</li>
    <li>python37Packages.scrapy-fake-useragent</li>
    <li>python37Packages.sslyze</li>
    <li>python37Packages.tld</li>
    <li>python37Packages.toggl-cli</li>
    <li>python37Packages.ufoprocessor</li>
    <li>python38Packages.Quandl (python38Packages.quandl)</li>
    <li>python38Packages.afdko</li>
    <li>python38Packages.booleanoperations</li>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.defcon</li>
    <li>python38Packages.django_extensions</li>
    <li>python38Packages.django_silk</li>
    <li>python38Packages.factory_boy</li>
    <li>python38Packages.faker</li>
    <li>python38Packages.flask-restx</li>
    <li>python38Packages.fontmath</li>
    <li>python38Packages.fontparts</li>
    <li>python38Packages.fontpens</li>
    <li>python38Packages.fonttools</li>
    <li>python38Packages.foundationdb51</li>
    <li>python38Packages.foundationdb52</li>
    <li>python38Packages.hyperkitty</li>
    <li>python38Packages.kazoo</li>
    <li>python38Packages.mailman-web</li>
    <li>python38Packages.minio</li>
    <li>python38Packages.monosat</li>
    <li>python38Packages.mutatormath</li>
    <li>python38Packages.nototools</li>
    <li>python38Packages.psautohint</li>
    <li>python38Packages.pytest-factoryboy</li>
    <li>python38Packages.pytest-randomly</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.rpy2</li>
    <li>python38Packages.scrapy-fake-useragent</li>
    <li>python38Packages.sslyze</li>
    <li>python38Packages.tld</li>
    <li>python38Packages.toggl-cli</li>
    <li>python38Packages.ufoprocessor</li>
    <li>python39Packages.afdko</li>
    <li>python39Packages.booleanoperations</li>
    <li>python39Packages.defcon</li>
    <li>python39Packages.django_extensions</li>
    <li>python39Packages.factory_boy</li>
    <li>python39Packages.faker</li>
    <li>python39Packages.flask-restx</li>
    <li>python39Packages.fontmath</li>
    <li>python39Packages.fontparts</li>
    <li>python39Packages.fontpens</li>
    <li>python39Packages.fonttools</li>
    <li>python39Packages.foundationdb51</li>
    <li>python39Packages.foundationdb52</li>
    <li>python39Packages.hyperkitty</li>
    <li>python39Packages.kazoo</li>
    <li>python39Packages.mailman-web</li>
    <li>python39Packages.minio</li>
    <li>python39Packages.monosat</li>
    <li>python39Packages.mutatormath</li>
    <li>python39Packages.nototools</li>
    <li>python39Packages.psautohint</li>
    <li>python39Packages.pytest-factoryboy</li>
    <li>python39Packages.pytest-randomly</li>
    <li>python39Packages.scrapy-fake-useragent</li>
    <li>python39Packages.sslyze</li>
    <li>python39Packages.tld</li>
    <li>python39Packages.ufoprocessor</li>
  </ul>
</details>

**Comment by Mic92** [comment]
Broken python packages are also broken on master:

✖ (Dependency failed) python3.7-foundationdb-6.0.18 from 2021-01-13 - https://hydra.nixos.org/build/135046768
✖ (Failed) python3.7-pylint-django-2.3.0 from 2021-01-11 - https://hydra.nixos.org/build/134824122
✖ (Dependency failed) python3.8-slither-analyzer-0.7.0 from 2021-01-13 - https://hydra.nixos.org/build/135031118

**Pull request merged**